classes/person_class.rb

```diff
@@ -1,19 +1,24 @@
 require_relative 'nameable'
+require_relative 'rental_class'
 
 class Person < Nameable
+  attr_reader :i
+  attr_accessor :name, :age, :rentals
+
   # Constructor with instance variables
   def initialize(name = 'Unknown', age = nil, parent_permission: true)
     super()
     @id = generate_id
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
   end
 
-  # Getters and setteres
-
-  attr_reader :i
-  attr_accessor :name, :age
+  def add_rental(rental)
+    @rentals.push(rental)
+    @rentals << rental
+  end
```

Comment:
- [ ] Dear @kerlos-bahaa, Just as mentioned in the `book` class, when we add a rental, instead of accepting a `rental` as a parameter, we need to accept the `book`, and `date` as a parameter and create the rental as a new one with the `person` being `self`, creating a new rental will add the rental to the `person.rentals` inside its initializer method. This way of doing this helps us preserve the association between `rental` and the `person`.

> something like `Rental.new(date, book, self)`

> Remember if we don't pass `self` as a parameter, there is no way of guaranteeing we are working with this `person`.

Reply:
This change has been done well